client/src/pages/recording-view/components/annotation/annotation-viewer.tsx

```diff
@@ -2,11 +2,64 @@
 // Copyright (c) 2023 Marc Lichtman
 // Licensed under the MIT License
 
-import React, { Fragment, useCallback, useMemo } from 'react';
-import { Layer, Rect, Text } from 'react-konva';
+import React, { Fragment, useEffect, useCallback, useMemo, useState } from 'react';
+import { Layer, Rect, Text, Group } from 'react-konva';
 import { TILE_SIZE_IN_IQ_SAMPLES } from '@/utils/constants';
 import { Annotation, SigMFMetadata } from '@/utils/sigmfMetadata';
 import { useSpectrogramContext } from '../../hooks/use-spectrogram-context';
+import { Html } from 'react-konva-utils';
+
+interface AnnotationLabelEditorProps {
+  labelText: string;
+  y: number;
+  x: number;
+  id: string;
+  setEditAnnotationLabelY: (y: number) => void;
+  setEditAnnotationLabelText: (text: string) => void;
+  setEditAnnotationLabelChanged: (flag: boolean) => void;
+}
+
+export const AnnotationLabelEditor = ({
+  labelText,
+  y,
+  x,
+  id,
+  setEditAnnotationLabelY,
+  setEditAnnotationLabelText,
+  setEditAnnotationLabelChanged,
+}: AnnotationLabelEditorProps) => {
+  const onKeyDown = (e) => {
+    if (e.key === 'Enter') {
+      setEditAnnotationLabelY(null);
+      if (e.target.value == '') {
+        setEditAnnotationLabelText(labelText);
+      } else {
+        if (labelText !== e.target.value) {
+          setEditAnnotationLabelText(e.target.value);
+          setEditAnnotationLabelChanged(true);
+        }
+      }
+    }
+  };
+  const top = y + 'px';
+  const left = x + 'px';
+  const msgTop = (y - 30).toString() + 'px';
+  return (
+    <Group>
```

Comment:
You don't need to do <Group> or <Html>

<>

</>

Will suffice

Reply:
I think I understand what you mean, and changed it. Please confirm.